src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -329,10 +373,11 @@ public static Type GetElementType(Type propertyType)
                             elementType = args[0];
                         }
                     }
-                    else
+
+                    if (elementType == null)
```

Comment:
There are 3 callers -- parentType is used to format the error message. I did add some comments to the method and changed to using multiple return statements instead of one which I think makes the code easier to follow.